src/RecoveryServices/RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs

```diff
@@ -152,18 +152,31 @@ public AzureWorkloadProviderHelper(ServiceClientAdapter serviceClientAdapter)
             string itemName,
             string vaultName,
             string resourceGroupName,
-            Action<CmdletModel.ItemBase, ProtectedItemResource> extendedInfoProcessor)
+            Action<CmdletModel.ItemBase, ProtectedItemResource> extendedInfoProcessor, string friendlyName = null)
         {
             List<ProtectedItemResource> protectedItemGetResponses =
                 new List<ProtectedItemResource>();
 
-            if (!string.IsNullOrEmpty(itemName))
+            if (!string.IsNullOrEmpty(itemName) || !string.IsNullOrEmpty(friendlyName))
             {
                 protectedItems = protectedItems.Where(protectedItem =>
                 {
                     Dictionary<CmdletModel.UriEnums, string> dictionary = HelperUtils.ParseUri(protectedItem.Id);
+
                     string protectedItemUri = HelperUtils.GetProtectedItemUri(dictionary, protectedItem.Id);
-                    return protectedItemUri.ToLower().Contains(itemName.ToLower());
+
+                    bool filteredByUniqueName = itemName != null && (protectedItemUri.ToLower().Contains(itemName.ToLower()) );
+                    bool filteredByFriendlyName = false;
+
+                    if (protectedItem.Properties.BackupManagementType == "AzureStorage" && protectedItem.Properties.WorkloadType == "AzureFileShare")
+                    {
+
+                        string protectedItemFriendlyName = (protectedItem.Properties as AzureFileshareProtectedItem).FriendlyName;
+                        filteredByUniqueName = filteredByUniqueName || ( itemName != null && protectedItemFriendlyName.ToLower() == itemName.ToLower() );
```

Comment:
Note that this means previous versions of the cmdlets, and previous scripts written for them will be broken.  This is not allowed for stable cmdlets

Reply:
The previous script will be broken anyway after service side changes get checked in, which is scheduled to be in sometime next week. This is to provide a way for customers to use this cmdlet for new file shares created. Note that the scripts won't be broken for existing file shares. Only for newly created file shares customers will be affected, for which they can install this version of powershell to unblock themselves.

Reply:
@markcowl : As discussed in the meeting, service can't take friendly name because it needs the unique identifier to retrieve the right file share and restore the data. We will publish the right usage of service APIs and PS cmdlets to make sure customers are aware of the right practice/principles.